### PR TITLE
Add a 'section' parameter to ModifyConfig

### PR DESCRIPTION
**Pull request by hughsie**
This means we do not have to encode it into the key name.

Breaks API, so unsuitable for backporting.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -742,67 +742,87 @@ fu_engine_reset_config(FuEngine *self, const gchar *section, GError **error)
...
+				       "DecompressDelay",
+				       "NeedsActivation",
+				       "NeedsReboot",
+				       "RegistrationSupported",
+				       "RequestDelay",
+				       "RequestSupported",
+				       "VerifyDelay",
+				       "WriteDelay",
+				       "WriteSupported",
+				       NULL};
```
Comment (superm1) [inline src/fu-engine.c:805]: to me this is begging for a vfunc to call into a plugin rather than storing in the daemon now.

**Review by superm1 — COMMENTED** [review]
I think bash completion needs to be updated too?

**Review by superm1 — DISMISSED** [review]

**Comment by superm1** [comment]
So given how nicely this works, should we just extend it to _all_ configuration options?  Let msr, thunderbolt, etc export their settings this way too?

**Comment by hughsie** [comment]
@superm1 I've overwritten your `trivial: add bash-completion for modify-config changes` -- sorry -- can you repush please.

**Pull request merged**